cpp/src/groupby/hash/groupby.cu

```diff
@@ -258,20 +288,45 @@ void compute_single_pass_aggs(table_view const& keys,
   bool skip_key_rows_with_nulls = keys_have_nulls and 
                                   include_null_keys == include_nulls::NO;
 
-  if (skip_key_rows_with_nulls) {
-    auto row_bitmask{bitmask_and(keys, rmm::mr::get_default_resource(), stream)};
-    thrust::for_each_n(rmm::exec_policy(stream)->on(stream),
-      thrust::make_counting_iterator(0), keys.num_rows(),
-      hash::compute_single_pass_aggs<true, Map>{
-        map, keys.num_rows(), *d_values, *d_sparse_table, d_aggs.data().get(),
-        static_cast<bitmask_type*>(row_bitmask.data())});
-  } else {
-    thrust::for_each_n(rmm::exec_policy(stream)->on(stream),
-      thrust::make_counting_iterator(0), keys.num_rows(),
-      hash::compute_single_pass_aggs<false, Map>{
-        map, keys.num_rows(), *d_values, *d_sparse_table, d_aggs.data().get(),
-        nullptr});
-  }
+  auto templated_compute_function = 
+    [skip_key_rows_with_nulls, target_nullable, source_nullable] () {
```

Comment:
This would be a little less messy by decoupling `skip_key_rows_with_nulls` from `target_nullable/source_nullable` and putting the `skip_key_rows_with_nulls` if/else inside `compute_single_pass_aggs_templated`.